gensim/corpora/dictionary.py

```diff
@@ -149,6 +149,7 @@ def doc2bow(self, document, allow_update=False, return_missing=False):
         token2id = self.token2id
         if allow_update or return_missing:
             missing = {w: freq for w, freq in iteritems(counter) if w not in token2id}
+            missing = OrderedDict(sorted(missing.items(), key=lambda x: (x[1], x[0])))
```

Comment:
It's really needed to sort items explicitly (everywhere)? If yes - please use `iteritems` instead of `.items()`

Reply:
This is the essential step because it introduces determinism in the assignment of ids and therefore consistency in the word-id mapping between executions (for a given dataset). However, `OrderedDict` can be omitted - see my comment regarding benchmarks for the implications of only using `sorted` and `dict`.